rg_instructor_analytics/views.py

```diff
@@ -1,37 +1,187 @@
+
+        As result return list of maps next format: {'is_active': Boolean, 'count': Integer}
+        Example of function result: [{'is_active': True, 'count': 10}, {'is_active': False, 'count': 2}]
+        """
+        return (
+            CourseEnrollment.history
+                            .filter(course_id=course_key, history_date__lt=date)
+                            .values('is_active')
+                            .annotate(count=Count('is_active')).order_by('is_active')
+        )
+
+    @staticmethod
+    def get_state_before(course_key, date):
+        """
+        Provide tuple with count of enroll and unenroll users.
+
+        For example - if database store 5 enrolled users and 2 unenrolled the result will be next: (5,-2)
+        """
+        stats = EnrollmentStatisticView.request_to_db_for_stats_before(course_key, date)
+        enrollment_count = 0
+        un_enrollment_count = 0
+        for s in stats:
+            if s['is_active']:
+                enrollment_count += s['count']
+            else:
+                un_enrollment_count -= s['count']
+        return enrollment_count, un_enrollment_count
+
+    @staticmethod
+    def get_state_in_period(course_key, from_date, to_date):
+        """
+        Provide list of tuples(date, is_active, count).
+
+        List contains next fields: date - day of activity, is_active - enrollment status,
+        count - the number of student with given activity in given day.
+        """
+        enrollment_query = (
+            CourseEnrollment.history
+                            .filter(course_id=course_key, history_date__range=(from_date, to_date))
+                            .annotate(date=RawSQL("select DATE(history_date)", (), output_field=DateField()))
+                            .values("date", "is_active")
+                            .annotate(count=Count('date'))
+                            .order_by('is_active').order_by('date')
+        )
+
+        return enrollment_query
+
+    @staticmethod
+    def get_statistic_per_day(from_timestamp, to_timestamp, course_key):
+        """
+        Provide statistic, which contains: dates in unix-time, count of enrolled users, unenrolled and total.
 
-class CalendarTabFragmentView(EdxFragmentView):
+        Return map with next keys: dates - store list of dates in unix-time, total - store list of active users
+        for given day (enrolled users - unenrolled),  enrol - store list of enrolled user for given day,
+        unenroll - store list of unenrolled user for given day.
+        """
+        from_date = datetime.fromtimestamp(from_timestamp)
+        to_date = datetime.fromtimestamp(to_timestamp)
+
+        enrollment_count, un_enrollment_count = EnrollmentStatisticView.get_state_before(course_key, from_date)
+        enrollments = EnrollmentStatisticView.get_state_in_period(course_key, from_date, to_date)
+
+        dates, counts_total, counts_enroll, counts_unenroll = [], [], [], []
```

Comment:
please write values tuple in an implicit way.

Reply:
done